pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Matrix Origin
```

Comment:
Why isn't this a regular projection?

Reply:
This `preinsertsecondaryindex` is copied from the current implementation of `preinsertunique` with one minor change mentioned below: 

```diff
	if colCount < 2 {
		msg := fmt.Sprintf("invalid secondary index column count in %s", catalog.IndexTableIndexColName)
		return process.ExecNext, moerr.NewInternalErrorNoCtx(msg)
	} else {
		vs := make([]*vector.Vector, colCount)
		for vIdx, pIdx := range secondaryColumnPos {
			vs[vIdx] = inputBat.Vecs[pIdx]
		}
-		vec, bitMap = util.SerialWithCompacted(vs, proc)
+               vec, bitMap = util.SerialWithoutCompacted(vs, proc)
	}
```

I believe @qingxinhome could give a better reasoning as to why we have not used a regular projection even for `unique index`.

Reply:
As per the discussion with @ouyuanning @nnsgmsone @qingxinhome , we will make this change in a separate PR.